packages/multi-test/src/custom_handler.rs

```diff
@@ -0,0 +1,60 @@
+use cosmwasm_std::{Addr, Api, Binary, BlockInfo, Empty, Storage};
+use mockall::automock;
+
+use anyhow::Result as AnyResult;
+
+use crate::AppResponse;
+
+/// Custom message handler trait. Implementator of this trait is mocking environment behaviour on
+/// given custom message.
+pub trait CustomHandler<ExecC = Empty, QueryC = Empty> {
+    fn execute(
+        &self,
+        api: &dyn Api,
+        storage: &mut dyn Storage,
+        block: &BlockInfo,
+        sender: Addr,
+        msg: ExecC,
+    ) -> AnyResult<AppResponse>;
+
+    fn query(
+        &self,
+        api: &dyn Api,
+        storage: &dyn Storage,
+        block: &BlockInfo,
+        msg: QueryC,
+    ) -> AnyResult<Binary>;
+}
+
+/// Simplified version of `CustomHandler` having only arguments which are not app internals - they
+/// are just discarded. Usefull for simpler mocking.
+#[automock(type QueryResult = Binary;)]
```

Comment:
I would feel happier removing this and just implementing the 2 needed methods, then mockall can be a dev-dependency only.

Reply:
In reality, we will likely have much larger blockchain specific logic in the custom handlers and just applying mocks here doesn't make too much sense usually.

Reply:
Basically this is the idea - specific blockchain impl just adds its own implementation. Auto-generated mock is just to make internal testing easier, but I will get rid of this in this place.